### ISSUE TITLE
Fire Field with stacked damage-over-time curses goes nuts with ticks

### ISSUE THREAD

**Issue by Marahin** [otland/forgottenserver#1704]
When a player receives a couple of DOT (damage over time) curses, and then someone casts a fire field beneath him, the ticks of the fire field go nuts, running every 200-600 ms I suppose. 

I am pretty sure it's not an intended behaviour or a feature, since it's possible to burn a player in literally two seconds. 
However, if it is indeed a feature and not a bug, how may I change this so the fire field doesn't tick the damage so fast? 

I attach a YouTube video where I have recorded how to reproduce the issue and how does it look like. 

https://youtu.be/WfUliAbItqQ

Thank you very much for the time spent reading this.

**Comment by Marahin** [comment]
Also, not sure but it might matter:
due to changes in the attack speed of vocations, I had to change EVENT_CREATURE_THINK_INTERVAL in src/creature.h, to this:
`#define EVENT_CREATURE_THINK_INTERVAL 50`

Maybe that's the cause of that weird ticks, but still: fire field without the DOT curses shown in the video works just fine and normal, as expected.

**Comment by BenDol** [comment]
I'm not able to reproduce this, are you using the standard spells from the data pack?

**Comment by Marahin** [comment]
Indeed, I've got the TFS 1.2 pulled around 23 days ago. I will happily share my source, which is only translated, no game mechanics changed (except going through people in PZ).

**Comment by ranisalt** [comment]
How come you pulled TFS 1.2 if it has [not been released yet](https://github.com/otland/forgottenserver/releases)?

**Comment by Marahin** [comment]
Correct me if I'm wrong, but I assumed so after seeing: 
`$ ./tfs
The Forgotten Server - Version 1.2
(...)
`
and seeing [THAT src/definitions.h:24](https://github.com/otland/forgottenserver/blob/master/src/definitions.h#L24):
`
#define STATUS_SERVER_VERSION "1.2"
`

**Comment by conde2** [comment]
Please do the following:

First try changing back the think interval to 1000 and see if the bug persist
#define EVENT_CREATURE_THINK_INTERVAL 1000

Second try changing 
	<item id="1487" article="a" name="fire field">
		<attribute key="type" value="magicfield" />
		<attribute key="decayTo" value="1488" />
		<attribute key="duration" value="120" />
		<attribute key="field" value="fire">
			<attribute key="damage" value="20" />
			<attribute key="ticks" value="10000" />
			<attribute key="count" value="7" />
			<attribute key="damage" value="10" />
		</attribute>
	</item>


TO

	<item id="1487" article="a" name="fire field">
		<attribute key="type" value="magicfield" />
		<attribute key="decayTo" value="1488" />
		<attribute key="duration" value="120" />
		<attribute key="field" value="fire">
			<attribute key="ticks" value="10000" />
			<attribute key="count" value="7" />
			<attribute key="damage" value="10" />
		</attribute>
	</item>

**Comment by Marahin** [comment]
@conde2 thank you for your answer. I haven't tried the first solution, since it would mean that I cannot change the attack speed of vocations, however I used the second suggestion and IT SEEMS that it fixed the issue. 
I will keep it updated if I find a way to reproduce it again, but it seems that removing second <damage> clause is a work-around in this issue.

**Comment by ranisalt** [comment]
You should set the creature intervalo to the GDC between your attack speed and 1000 for the best balance. No need to be as low as 50 if your attack speed is, say, 1500
